agent/xds/jwt_authn.go

```diff
@@ -84,67 +62,120 @@ func makeJWTAuthFilter(pCE map[string]*structs.JWTProviderConfigEntry, intention
 
 	cfg := &envoy_http_jwt_authn_v3.JwtAuthentication{
 		Providers: providers,
-		Rules:     rules,
+	}
+	// only add rules if any of the existing providers are referenced by intentions
+	if len(jwtRequirements) > 0 {
+		cfg.Rules = []*envoy_http_jwt_authn_v3.RequirementRule{
+			{
+				Match: &envoy_route_v3.RouteMatch{
+					PathSpecifier: &envoy_route_v3.RouteMatch_Prefix{Prefix: "/"},
+				},
+				RequirementType: makeJWTRequirementRule(andJWTRequirements(jwtRequirements)),
+			},
+		}
 	}
 	return makeEnvoyHTTPFilter(jwtEnvoyFilter, cfg)
 }
 
-func collectJWTAuthnProviders(i *structs.Intention) []*jwtAuthnProvider {
-	var reqs []*jwtAuthnProvider
+func makeJWTRequirementRule(r *envoy_http_jwt_authn_v3.JwtRequirement) *envoy_http_jwt_authn_v3.RequirementRule_Requires {
+	return &envoy_http_jwt_authn_v3.RequirementRule_Requires{
+		Requires: r,
+	}
+}
 
-	if i.JWT != nil {
-		for _, prov := range i.JWT.Providers {
-			reqs = append(reqs, &jwtAuthnProvider{Provider: prov, ComputedName: makeComputedProviderName(prov.Name, nil, 0)})
+// andJWTRequirements combines list of jwt requirements into a single jwt requirement.
+func andJWTRequirements(reqs []*envoy_http_jwt_authn_v3.JwtRequirement) *envoy_http_jwt_authn_v3.JwtRequirement {
+	switch len(reqs) {
+	case 0:
+		return anyJWTRequirement()
```

Comment:
Why would we need to configure an empty `requires_any`? What happens when the `requires_any` list is empty in the JWT auth filter?

Reply:
in our current implementation, [we don't really get to this as there is a check ](https://github.com/hashicorp/consul/pull/18062/files#diff-08190539d43b885f27cd79de992a11df3bcceb6e6304892677179d7985a25679R67) to call this only when there are requirements.

This is more of a safety case than anything. It will probably throw an error for our current use case: `JwtRequirementOrListValidationError.Requirements: value must contain at least 2 item(s))`. I will update this to return nil instead